phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewIT.java

```diff
@@ -372,6 +378,31 @@ public void testViewAndTableAndDrop() throws Exception {
         // drop table cascade should succeed
         conn.createStatement().execute("DROP TABLE " + fullTableName + " CASCADE");
         
+        validateViewDoesNotExist(conn, fullViewName1);
+        validateViewDoesNotExist(conn, fullViewName2);
+
+    }
+    
+    @Test
+    public void testRecreateDroppedTableWithChildViews() throws Exception {
```

Comment:
These new tests are good. These are testing that the left over metadata doesn't impact the re-creation of a table since we don't make the RPC to delete views when a base table is dropped, right? Do you think there'd be any issues if part of the rows for a view were there (i.e. say that the create view failed, but some of the rows were written)? Might be good to have a test like this - you could set it up by using HBase APIs to manually delete some rows of a view.

Reply:
We write the parent->child link first, then if the table uses column encoding we update the encoded column qualifiers on the parent table, and finally use mutateRowsWithLocks to write the view metadata atomically. 
We ignore views that can't be found (in case writing the child view metadata fails). 
If the metadata write fails and the table uses column encoding then we will lose a few column qualifiers. 
I'll add a test for this.